content/en/docs/how-tos/use-registries-in-build-farm.md

```diff
@@ -1,112 +1,243 @@
+`OpenShift` organization.
 
-If some robot is required to log into the central CI registry, we need to use a service account: Choose a new namespace, and create a pull request to add a folder with the same name in [release/clusters/app.ci](https://github.com/openshift/release/tree/master/clusters/app.ci). The folder should contain the namespace manifest.
+## How do I get a token for programmatic access to the central CI registry?
 
-After the pull request is merged, the manifests in the folder will be applied automatically to `app.ci`. We can use `sa/builder` which [Openshift creates by default for any namespace and is allowed to pull images from any imagestream in the namespace](https://docs.openshift.com/container-platform/4.6/authentication/using-service-accounts-in-applications.html#default-service-accounts-and-roles_using-service-accounts). [Openshift also creates secrets for each service account for the OpenShift Container Registry](https://docs.openshift.com/container-platform/4.6/authentication/using-service-accounts-in-applications.html#service-accounts-overview_using-service-accounts). The secret name could be found at `sa.imagePullSecrets`.
+If you're developing an integration with the central CI registry, an OpenShift `ServiceAccount` should be used. Write a
+pull request to the [`openshift/release`](https://github.com/openshift/release) repository that adds a new directory under
+the `release/clusters/app.ci/registry-access` directory. In this directory, provide an `OWNERS` file to allow your team
+to make changes to your manifests and an `admin_manifest.yaml` file that creates your `ServiceAccount` and associated
+RBAC:
 
-We can then get the token of, e.g., `SA/builder` in `namespace/ci`:
+```yaml
+# this is the Namespace in which your ServiceAccount will live
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations:
+    openshift.io/description: Automation ServiceAccounts for MyProject
+    openshift.io/display-name: MyProject CI
+  name: my-project
+---
+# this is the ServiceAccount whose credentials you will use
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: image-puller
+  namespace: my-project
+---
+# this grants your ServiceAccount rights to pull images
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: my-project-image-puller-binding
+  # the namespace from which you will pull images
+  namespace: ocp
+roleRef:
+  kind: ClusterRole
+  apiGroup: rbac.authorization.k8s.io
+  name: system:image-puller
+subjects:
+  - kind: ServiceAccount
+    namespace: my-project
+    name: image-puller
+---
+# the Group of people who should be able to manage this ServiceAccount
+kind: Group
+apiVersion: v1
+metadata:
+  name: my-project-admins
+users:
+  # these names are GitHub usernames
+  - bob
+  - tracy
+  - jim
+  - emily
+---
+# this allows the group of people admin access to the Namespace
+kind: RoleBinding
```

Comment:
This will end up being used on app.ci, can we not suggest Clusteradmin and instead just the `imagePuller`?

Reply:
We need them to be able to get the tokens for their service account. If you want to find the RBAC for that, sure. This namespaced binding of cluster-admin will not give them anything they shouldn't have in their namespace IMO

Reply:
It will give them perms to create workloads and they are not supposed to do that on app.ci. We essentially need a role (could pre-create it) that gives all access to secrets, serviceaccounts and configmaps but nothing else

Reply:
(This is also the current YAML people use in `api.ci` for this purpose)